src/nr_openai_observability/bedrock.py

```diff
@@ -192,25 +192,33 @@ def handle_bedrock_create_completion(
         monitor.record_event(summary, SummaryEventName)
         monitor.record_event(transaction_begin_event, TransactionBeginEventName)
 
-    except Exception as error:
-        build_completion_summary_for_error(error, **kwargs)
+    except BaseException as error:
+        build_bedrock_completion_summary_for_error(error, completion_id)
         raise error
 
 
 def handle_bedrock_embedding(result, contents, time_delta, **kwargs):
     embedding_id = str(uuid.uuid4())
-    response_body = json.loads(contents)
     input_body = json.loads(kwargs.get("body"))
+    model = kwargs.get("modelId")
     request_id = None
     if None != result.get("ResponseMetadata") and None != result.get(
         "ResponseMetadata"
     ).get("RequestId"):
         request_id = result.get("ResponseMetadata").get("RequestId")
 
+    text = ""
+
+    if "titan-embed" in model:
+        text = input_body.get("inputText")
+    elif "cohere.embed" in model:
+        for t in input_body.get("texts"):
+            text += f"{t}\n"
```

Comment:
Just double-checking -- Is this texts API actually providing multiple embeddings in a batch operation? If that were the case, would we want one event per embedding or one event for the whole batch operation?

Reply:
The input is a list of strings with various restrictions on both the length of each string and the number of entries in the list. For each input string, the output contains a list of vectors; should be one for each input. The output is not stored at all within the embedding and we also do not have token information for any of this with Bedrock.

In theory, we could create 1 event per entry in the input. It is very unclear how this will be used, so for now I am just merging the input text and grabbing the first 4kb to be stored. I believe we are lacking use cases on how this information will be used and that would help clarify what the correct approach would be.